parser.cpp

```diff
@@ -0,0 +1,32 @@
+#include "redis.hpp"
+
+vector<string> parse(string line) {
+  size_t iterator = 0;
+  size_t start = 0;
+  vector<string> args;
+  if (line == "") {
```

Comment:
Instead of returning a vector with an empty first argument when the input is empty, I would rather you return an empty vector. I know you did it to simplify the check in the `function_mapper`, but the intent might not be clear and the caller might not expect it. If you return an empty vector, you signal to the caller that there are no parsed arguments and that the result should be handled at the call site (in this case it's `function_mapper`).